Update price-bundle.js so that correct tier price is calculated while displaying in bundle product

adarshkhatri opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->
By default tier price is sorted by "price_id". With this tier price calculation while displaying in bundle product is wrong. Need to sort tier price by "price_qty".

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. magento/magento2#21467: Tier price of simple item not working in Bundle product

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. Create tier price as per described in above issue

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

adarshkhatri committed:
Update price-bundle.js

By default tier price is sorted by "price_id". With this tier price calculation while displaying in bundle product is wrong. Need to sort tier price by "price_qty".
Issue created: https://github.com/magento/magento2/issues/21467

magento-engcom-team commented:
Hi @adarshkhatri. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento/magento2?pullRequest=21469) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=21469) before we can accept your contribution.<br/>

adarshkhatri committed:
Update price-bundle.js

By default tier price is sorted by "price_id". With this tier price calculation while displaying in bundle product is wrong. Need to sort tier price by "price_qty".

adarshkhatri committed:
Update price-bundle.js

VladimirZaets commented on app/code/Magento/Bundle/view/base/web/js/price-bundle.js:
```diff
@@ -375,6 +375,17 @@ define([
             lowest = false;
+        
+        //tiers is undefined when options has only one option 
+		if (undefined == tiers) {
```
Hi @adarshkhatri. Thanks for collaboration. 
According to [Magento Technical Guidelines](https://devdocs.magento.com/guides/v2.2/coding-standards/technical-guidelines.html) we should use `===` instead of `==`. 

In this case, you can make only a simple check `if (tiers)` or if you need to compare value directly to undefined you can use underscore.js method _.isUndefined

adarshkhatri replied:
oh! thanks, will update accordingly.

VladimirZaets commented on app/code/Magento/Bundle/view/base/web/js/price-bundle.js:
```diff
@@ -375,6 +375,17 @@ define([
+        
+        //tiers is undefined when options has only one option 
+		if (undefined == tiers) {
+		    var firstKey = Object.keys(optionConfig)[0];
```
According to [Magento Coding Standard](https://devdocs.magento.com/guides/v2.2/coding-standards/code-standard-javascript.html) all variable declaration should be at the start in the function.

adarshkhatri replied:
thanks, have updated accordingly.

adarshkhatri committed:
Update price-bundle.js

adarshkhatri committed:
Update price-bundle.js

adarshkhatri committed:
Update price-bundle.js

VladimirZaets commented:
@adarshkhatri thanks

magento-engcom-team commented:
Hi @VladimirZaets, thank you for the review.
[ENGCOM-4507](https://jira.corp.magento.com/browse/ENGCOM-4507) has been created to process this Pull Request

magento-engcom-team commented:
@adarshkhatri thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

VasylShvorak commented:
✔️ QA passed
